axelor-base/src/main/java/com/axelor/apps/base/service/ImportExportTranslationServiceImpl.java

```diff
@@ -0,0 +1,357 @@
+package com.axelor.apps.base.service;
+
+import com.axelor.apps.base.db.ImportExportTranslation;
+import com.axelor.apps.base.db.ImportExportTranslationHistory;
+import com.axelor.apps.base.db.repo.ImportExportTranslationHistoryRepository;
+import com.axelor.apps.base.db.repo.ImportExportTranslationRepository;
+import com.axelor.db.JPA;
+import com.axelor.inject.Beans;
+import com.axelor.meta.MetaFiles;
+import com.axelor.meta.db.MetaTranslation;
+import com.axelor.meta.db.repo.MetaTranslationRepository;
+import com.axelor.utils.helpers.file.CsvHelper;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.persistence.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ImportExportTranslationServiceImpl implements ImportExportTranslationService {
+
+  protected MetaTranslationRepository metaTranslationRepository;
+  protected ImportExportTranslationRepository importExportTranslationRepository;
+  protected ImportExportTranslationHistoryRepository importExportTranslationHistoryRepository;
+
+  public static final int LIMIT = 50;
+
+  protected int errorNumber = 0;
+  protected int countRecords = 0;
+
+  @Inject
+  public ImportExportTranslationServiceImpl(
+      MetaTranslationRepository metaTranslationRepository,
+      ImportExportTranslationRepository importExportTranslationRepository,
+      ImportExportTranslationHistoryRepository importExportTranslationHistoryRepository) {
+    this.metaTranslationRepository = metaTranslationRepository;
+    this.importExportTranslationRepository = importExportTranslationRepository;
+    this.importExportTranslationHistoryRepository = importExportTranslationHistoryRepository;
+  }
+
+  /**
+   * This method created an exported translation csv file for the input ImportExportTranslation
+   * object.
+   *
+   * @param importExportTranslation
+   * @throws IOException
+   */
+  @Transactional
```

Comment:
You should not open a transaction here and do some JPA.clear in a method inside
I think your JPA.clear is useless because you just fetch some elements, it's more useful when we save a lot of things